gcc/config/riscv/riscv-sr.c

```diff
@@ -171,6 +171,42 @@ riscv_sr_match_epilogue (void)
   return start;
 }
 
+/* If the function has stack incomming arguments, find all the uses of sp
+   between its prologue and its epilogue and update their offset  */
+static void
+update_sp_use (rtx_insn *prologue, rtx_insn *epilogue)
+{
+  rtx_insn *insn;
+
+  /* Skip if the function has no stack incomming arguments */
+  if (crtl->args.size == 0)
+    return;
+
+  for (insn = NEXT_INSN (prologue); insn != epilogue; insn = NEXT_INSN (insn))
+    if (INSN_P (insn) && GET_CODE (PATTERN (insn)) == SET)
+      {
+	rtx pat_src = SET_SRC (PATTERN (insn));
+
+	/* case assigning sp value */
+	if (GET_CODE (pat_src) == PLUS && REG_P (XEXP (pat_src, 0))
+	    && REGNO (XEXP (pat_src, 0)) == STACK_POINTER_REGNUM)
+	  {
+	    INTVAL (XEXP (pat_src, 1)) -= TARGET_RVE ? 12 : 16;
+	    continue;
+	  }
+
+	/* case loading stack incoming arguments  */
+	if (GET_CODE (pat_src) == SIGN_EXTEND
+	    || GET_CODE (pat_src) == ZERO_EXTEND)
+	  pat_src = XEXP (pat_src, 0);
+
+	if (MEM_P (pat_src) && GET_CODE (XEXP (pat_src, 0)) == PLUS
+	    && REG_P (XEXP (XEXP (pat_src, 0), 0))
+	    && REGNO (XEXP (XEXP (pat_src, 0), 0)) == STACK_POINTER_REGNUM)
+	  INTVAL (XEXP (XEXP (pat_src, 0), 1)) -= TARGET_RVE ? 12 : 16;
```

Comment:
Doesn't verify that XEXP (XEXP (pat_src, 0), 1) is a CONST_INT before modifying it.